languagetool-language-modules/de/src/main/resources/org/languagetool/rules/de/grammar.xml

```diff
@@ -30502,15 +30502,15 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
                     <token>.</token>
                     <token regexp="yes">\d\d\d\d</token>
                 </pattern>
-                <filter class="org.languagetool.rules.de.DateCheckFilter" args="year:\9 month:\7 day:\5 weekDay:\1"/>
+                <filter class="org.languagetool.rules.de.DateCheckFilter" args="year:9 month:7 day:5 weekDay:1"/>
                 <message>Das Datum \5.\7.\9 fällt nicht auf einen {day}, sondern auf einen {realDay}.</message>
-                <example correction="">Termin: <marker>Sonntag, 23.08.2014</marker></example>
-                <example correction="">Termin: <marker>Sonntag, 23.8.2014</marker></example>
-                <example correction="">Termin: <marker>So, 23.08.2014</marker></example>
-                <example correction="">Termin: <marker>So., 23.08.2014</marker></example>
-                <example correction="">Termin: <marker>Sonntag, den 23.08.2014</marker></example>
-                <example correction="">Termin: <marker>So, den 23.08.2014</marker></example>
-                <example correction="">Termin: <marker>So., den 23.08.2014</marker></example>
+                <example correction="Samstag, 23|Sonntag, 24">Termin: <marker>Sonntag, 23</marker>.08.2014</example>
+                <example correction="Samstag, 23|Sonntag, 24">Termin: <marker>Sonntag, 23</marker>.8.2014</example>
+                <example correction="Samstag, 23|So, 24">Termin: <marker>So, 23</marker>.08.2014</example>
+                <example correction="Samstag., 23|So., 24">Termin: <marker>So., 23</marker>.08.2014</example>
```

Comment:
When the weekday is spelled out (Samstag), we don't need the dot (only the comma: Samstag,) but when the weekday is shortened (So.) we need the dot (and the comma: So.,)

Reply:
Thank you. I think I can fix it, without making it too complicated.